ginga/misc/plugins/SaveImage.py

```diff
@@ -0,0 +1,477 @@
+        for key in func():
+            iminfo = self.chinfo.get_image_info(key)
+            path = iminfo.get('path')
+            idx = iminfo.get('idx')
+            t = iminfo.get('time_modified')
+
+            if path is None:  # Special handling for generated buffer, eg mosaic
+                infile = key
+                is_fits = True
+            else:
+                infile = os.path.basename(path)
+                infile_ext = os.path.splitext(path)[1]
+                infile_ext = infile_ext.lower()
+                is_fits = False
+
+                if 'fit' in infile_ext:
+                    is_fits = True
+
+            # Only list FITS files unless it is Ginga generated buffer
+            if not is_fits:
+                continue
+
+            # Only list modified buffers
+            if mod_only and t is None:
+                continue
+
+            # More than one ext modified, append to existing entry
+            if infile in treedict:
+                if t is not None:
+                    treedict[infile].extlist.add(idx)
+                    elist = sorted(treedict[infile].extlist)
+                    treedict[infile].MODEXT = ';'.join(
+                        map(self._format_extname, elist))
+
+            # Add new entry
+            else:
+                if t is None:
+                    s = ''
+                    extlist = set()
+                else:
+                    s = self._format_extname(idx)
+                    extlist = set([idx])
+                treedict[infile] = Bunch.Bunch(
+                    IMAGE=infile, MODEXT=s, extlist=extlist, path=path)
+
+        self.treeview.set_tree(treedict)
+
+        # Resize column widths
+        n_rows = len(treedict)
+        if n_rows == 0:
+            self.w.status.set_text('Nothing available for saving')
+        elif n_rows < self.settings.get('max_rows_for_col_resize', 5000):
+            self.treeview.set_optimal_column_widths()
+            self.logger.debug('Resized columns for {0} row(s)'.format(n_rows))
+
+        return
+
+    def _check_chinfo(self):
+        """Special handling for when plugin is started from global plugin
+        menu.
+
+        This locks the plugin to the active channel when it starts.
+        To change the associated channel in global plugin space, user
+        needs to manual close and reload the plugin. Doing this
+        dynamically inside redo() messes up the behavior in
+        local plugin space.
+
+        """
+        self.chinfo = self.fv.get_channelInfo()
+
+        if self.chinfo is None:
+            raise ValueError('No channel available')
+
+        self.chname = self.chinfo.name
+
+    def keys_from_datasrc(self):
+        """Yield back keys for image listing from Ginga's data cache."""
+        for key in self.chinfo.datasrc.sortedkeys:
+            yield key
+
+    def keys_from_contents(self):
+        """Yield back keys for image listing  from ``Contents`` global plugin.
+        """
```

Comment:
It looks like you have designed the plugin to run either locally or globally.  If invoked locally, then it would make sense to show only the list of images in the channel it is associated with--user should not be confused if they understand the general difference between local/global. If invoked globally, or we are making the decision to allow any image to be saved then it might be good to show which channel the image came from.

One simple way would be to simply prepend a prefix when throwing the image name in the tree like:
`Image/u40x010hm_c0h.fits` or `Image: u40x010hm_c0h.fits`.